pkg/utils/etcdutil/etcdutil.go

```diff
@@ -461,6 +461,8 @@ func (lw *LoopWatcher) initFromEtcd(ctx context.Context) int64 {
 		watchStartRevision int64
 		err                error
 	)
+	ctx, cancel := context.WithTimeout(ctx, lw.loadTimeout)
+	defer cancel()
 	ticker := time.NewTicker(defaultLoadFromEtcdRetryInterval)
 	defer ticker.Stop()
```

Comment:
Since it has a timeout, we better initialize it later.

```suggestion
	ticker := time.NewTicker(defaultLoadFromEtcdRetryInterval)
	defer ticker.Stop()
	ctx, cancel := context.WithTimeout(ctx, lw.loadTimeout)
	defer cancel()
```